replica/database.cc

```diff
@@ -2577,6 +2577,10 @@ future<> database::truncate_table_on_all_shards(sharded<database>& sharded_db, s
     if (!sharded_db.local().get_config().auto_snapshot()) {
         with_snapshot = false;
     }
+    if (with_snapshot && !table_shards->get_storage_options().is_local_type()) {
+        dblog.warn("Skipping snapshot for {}.{} because it's not of local type", s->ks_name(), s->cf_name());
+        with_snapshot = false;
```

Comment:
‎The patch is good, it's certainly better than failing the deletion because of an on_internal_error.

If the auto_snapshot option was typically given manually, I would ask that when a table is created with auto_snapshot=true and s3 storage we should print a CQL warning (not just a log message, and not just when the deletion happened - that's too late). But since auto_snapshot **defaults** to true and it's not really clear why we want it, I think this is fine.